### ISSUE TITLE
VIRA-213: Autoload of Vira Server/Project

### ISSUE THREAD

**Issue by nathanaelcunningham** [n0v1c3/vira#70]
How exactly does `When you're in a git repo, vira will auto-load your pre-defined settings by matching the local repo name from file path.` work? I have a vira_projects.json added to the default path following the readme, but whenever i try to look at issues i have to select the server first.

**Comment by mikeboiko** [comment]
vira tries a couple of different methods in this order.
1. Try to get local git repo using `git rev-parse --show-toplevel`
2. Get last folder name using `pwd`

See code snippet from `load_project_config()` in `vira/python/Vira/vira_api.py`:

```
            repo = run_command('git rev-parse --show-toplevel')['stdout'].strip()
            if not self.vira_projects.get(repo):
                repo = repo.split('/')[-1]
            if not self.vira_projects.get(repo):
                repo = run_command('pwd')['stdout'].strip()
            if not self.vira_projects.get(repo):
                repo = repo.split('/')[-1]
            if not self.vira_projects.get(repo):
                repo = '__default__'
                if not self.vira_projects.get('__default__'):
                    return
```

**Comment by n0v1c3** [comment]
Hello @nathanaelcunningham the main goal was put in by @mikeboiko. When you are inside a folder and run `vira` ie `connect` to the server it will check that list that @mikeboiko mentioned above to see if there is a best fit otherwise fall to the `__default__` settings or NO filter if there is no file at all. (Here simply the `vira_servers.json/yaml` file will be needed.

If you are inside `vim` once `vira` is already running a filter basically already exists this is where the commands like `ViraFiterEdit` and `ViraFilterProjects` will be useful. You can use these commands to edit the filters "on the fly" while using the "ViraFilterReset" to get back to NO filters other than the server you are connected to.

It looks like I may need to do a patch for the highlighting in the `ViraFilterProjects` menu but here are the key mappings and still appear to work:

- [ ] `D` remove all selections, confirm and close menu
- [ ] `s` select a single item for the list
- [ ] `S` add all items to the list
- [ ] `u` unselect a single item from the list
- [ ] `U` unselect all items from the list
- [ ] `<cr>` confirm selection and close menu

**Comment by n0v1c3** [comment]
There is also a bit better breakdown of the json format in [#41](https://github.com/n0v1c3/vira/issues/41) I will probable just update that at the same time and just call this one a bit easier for others.

**Comment by n0v1c3** [comment]
As I am reading to begin cleanup I can see som confusion is possible. I will try to make a simple as can be for here is `json` and here is `yaml` with a list bellow for what is each row. This way we can have simply ONE section for "here is a file and how you can make a copy for what you need it to do for you"

I personally use the "ViraFilterEdit" command a lot as I can simply take what I make from there and put it inside my `vira_projects` file if I want to attach it to a directory.

**Comment by n0v1c3** [comment]
https://stackoverflow.com/questions/62642858/captcha-challenge-on-login-with-python-jira

**Comment by n0v1c3** [comment]
```
{  
  "__default__": {
    "server": "https://jira.testing.ca"
  },
  "__n0v1c3__": {
    "server": "https://n0v1c3.atlassian.net"
  },
  "__testing__": {
    "server": "https://jira.testing.ca"
  },
  "vira": {
    "template": "__default__",
    "filter": {
      "project": ["VIRA"],
      "assignee": ["travis"],
      "fixVersion": ["0.5.0"]
    }
  },
  "vim": {
    "server": "https://jira.testing.ca",
    "filter": {
      "project": "VIM"
    }
  },
  "workProject": {
    "server": "https://jira.work.net",
    "filter": {
      "project": ["HARD", "EASY", "MONEY"],
      "fixVersion": ["0.0.3", "0.0.4", "0.0.5"]
    }
  }
}
```

**Comment by n0v1c3** [comment]
@nathanaelcunningham give a test on the `VIRA-207` branch if you have a chance. I think that you had only one server and that was a bug in my code.

**Comment by nathanaelcunningham** [comment]
@n0v1c3 That branch is working. I tested the projects file with just the default server and that worked. Ill try project specific server setup in a bit.

**Comment by n0v1c3** [comment]
I have a feeling this one has floated past me for a little while now as my test we all on multiple servers setup. One more test on a list that is being created. Thanks!

I did start with a `README` cleanup so I will probably do a quick wrapup of that and push it to master for the rest of the people. It appears to work fine on `vim` and `nvim`. Multiple servers still appears to work and should just open that menu if a `__default__` is not defined.

I am sure some deeper testing will find something wrong and I know I took a sloppy way out but I am not worried with were my code is going for `VIRA 0.5.0`. The menus are going to completely change how they load and this will be affected.

**Comment by n0v1c3** [comment]
@nathanaelcunningham and @mikeboiko I daringly moved it to the `master` branch so lets see how that goes for everyone out there. It did not seem to break anythiny I had set up and only made it better so I cannot see to many complaints coming.

Thanks @nathanaelcunningham, I know that I started with the `README` and your config but I am sure there are other ways to keep making it better as I don't think I will ever get to finish VIRA-213. (It is a `README` based issue and here we are changing the login process).
